dataland-frontend/tests/e2e/specs/search-company-framework-data/ViewMyDatasets.ts

```diff
@@ -0,0 +1,155 @@
+          return `.p-tabview-header[data-index="${tabIndex}"]`;
+        }
+
+        /**
+         * Validates the tab bar identified by the input
+         *
+         * @param activeTabIndex number identifying the tab bar
+         */
+        function validateTabBar(activeTabIndex: number): void {
+          cy.get(getTabSelector(0)).should("have.text", "AVAILABLE DATASETS");
+          cy.get(getTabSelector(1)).should("have.text", "MY DATASETS");
+          const inactiveTabIndex = (activeTabIndex + 1) % 2;
+          cy.get(getTabSelector(activeTabIndex)).should("have.class", "p-highlight");
+          cy.get(getTabSelector(inactiveTabIndex)).should("not.have.class", "p-highlight");
+        }
+
+        it("Check page content", function () {
+          cy.ensureLoggedIn(uploader_name, uploader_pw);
+          cy.visitAndCheckAppMount("/companies");
+          validateTabBar(0);
+          cy.wait(3000);
+          const companiesInterceptAlias = "companies";
+          cy.intercept("**/api/companies*onlyCurrentUserAsUploader*").as(companiesInterceptAlias);
+          cy.get(getTabSelector(1)).click();
+          cy.wait(`@${companiesInterceptAlias}`, { timeout: Cypress.env("long_timeout_in_ms") as number });
+          cy.url().should("contain", "/datasets");
+          validateTabBar(1);
+
+          cy.get(newDatasetButtonSelector).should("exist");
+
+          const expectedHeaders = ["COMPANY", "DATA FRAMEWORK", "SUBMISSION DATE"];
+          expectedHeaders.forEach((value) => {
+            cy.get(`table th:contains(${value})`).should("exist");
+          });
+          const unexpectedHeaders = ["YEAR"];
+          unexpectedHeaders.forEach((value) => {
+            cy.get(`table th:contains(${value})`).should("not.exist");
+          });
+          cy.get("th").each((element) => {
+            if (!expectedHeaders.includes(element.text())) {
+              expect(element.html()).to.contain("<input");
+            }
+          });
+        });
+
+        it("Check if new dataset button leads to choose company page and if the back button works as expected", () => {
+          cy.ensureLoggedIn(uploader_name, uploader_pw);
+          cy.visit("/datasets");
+          validateTabBar(1);
+          cy.get(newDatasetButtonSelector).click();
+          cy.url().should("contain", "/companies/choose");
+          cy.get("[title='back_button']").click();
+          cy.url().should("contain", "/datasets");
+        });
+
+        it("Check that a user who has no dataset associated with him has no table displayed", () => {
+          cy.ensureLoggedIn(admin_name, admin_pw);
+          cy.visit("/datasets");
+          cy.get(newDatasetButtonSelector).should("exist");
+          cy.get("table").parent().parent().parent().should("have.class", "hidden");
+        });
+
+        it("Check that a user who has no upload permission has no new dataset button displayed", () => {
+          cy.ensureLoggedIn(admin_name, admin_pw);
```

Comment:
admin role is not allowed to upload data?

Reply:
I'm confused. Why does this test work? Investigating...

Reply:
Could you find out why this test succeeded even though it should not have succeeded @fsoffel ?

Reply:
Done. Racing condition, it worked because stuff was not loaded yet. Therefore a non exsistence test did not fail.

Reply:
If its fine for you Emanuel, i would resolve the conversion then.

Reply:
Ok yes it is fine for me.